sympy/core/tests/test_basic.py

```diff
@@ -36,6 +36,18 @@ def test_equality():
     assert not(Basic() == 0)
 
 
+def test_compare_issue_9464_a():
+    assert(Basic(1).compare(Basic(None)) > 0)
+
+
+def test_compare_issue_9464_b():
+    assert(Basic(None).compare(Basic(1)) < 0)
+
+
+def test_compare_issue_9464_c():
+    assert(Basic(None).compare(Basic(None)) == 0)
+
+
```

Comment:
I wasn't sure that comparison of "None" with "not None" also causes exception, because in my test program mentioned in issue #9464, I saw only crash caused by compare "None" with "None". So I decided to demonstrate, that all 3 comparisons produce exception, and I really need to implement suggested modification - full check for None with something else. That's why I did 2 separate commits in my project fork.

On the other hand, you are also right and possibly I have to add 3-rd commit to my pull request where merge 3 tests into single one and repeat request?